src/vmm/src/controller.rs

```diff
@@ -57,21 +58,23 @@ impl VmmController {
     #[cfg(target_arch = "x86_64")]
     pub fn send_ctrl_alt_del(&mut self) -> ActionResult {
         self.vmm
+            .lock()
+            .unwrap()
             .send_ctrl_alt_del()
             .map_err(VmmActionError::InternalVmm)
     }
 
     /// Stops the inner Vmm and exits the process with the provided exit_code.
     pub fn stop(&mut self, exit_code: i32) {
-        self.vmm.stop(exit_code)
+        self.vmm.lock().unwrap().stop(exit_code)
     }
 
     /// Creates a new `VmmController`.
     pub fn new(
         epoll_context: EpollContext,
         event_manager: EventManager,
         vm_resources: VmResources,
-        vmm: Vmm,
+        vmm: Arc<Mutex<Vmm>>,
```

Comment:
As we discussed offline, I think we shouldn't wrap the whole Vmm object in an Arc<Mutex>. We can change this in a future PR by either deciding on dropping Arc<Mutex> for all implementors of `EpollHandler` or we can just lock parts of the Vmm in an Arc<Mutex>.

Reply:
I would like to avoid wrapping the Vmm in an Arc<Mutex> as well, but as of now I have no better alternative.

We can refactor this later if we come up with a better solution. Right now, I think removing `EpollContext` is the highest priority.

Reply:
I can add an issue and we can think about a better solution after we finish with the decoupling. What do you say?

Reply:
👍